plenum/server/consensus/ordering_service.py

```diff
@@ -1,9 +1,173 @@
+        self._bus.subscribe(LedgerSyncStatus, self.set_is_synced)
+        self._bus.subscribe(ParticipatingStatus, self.set_is_participating)
+
+    def process_preprepare(self, pre_prepare: PrePrepare, sender: str):
+        result, reason = self._validate(pre_prepare)
+        if result == PROCESS:
+            self._process_preprepare(pre_prepare, sender)
+        return result
+
+    def process_prepare(self, prepare: Prepare, sender: str):
+        result, reason = self._validate(prepare)
+        if result == PROCESS:
+            self._process_prepare(prepare, sender)
+        return result
+
+    def process_commit(self, commit: Commit, sender: str):
+        result, reason = self._validate(commit)
+        if result == PROCESS:
+            self._process_commit(commit, sender)
+        return result
+
+    def _process_preprepare(self, pre_prepare: PrePrepare, sender: str):
+        pass
+
+    def _process_prepare(self, prepare: Prepare, sender: str):
+        pass
+
+    def _process_commit(self, commit: Commit, sender: str):
+        pass
+
+    @property
+    def view_no(self):
+        return self._data.view_no
+
+    @property
+    def legacy_last_prepared_sertificate(self):
+        """
+        We assume, that prepared list is an ordered list, and the last element is
+        the last quorumed Prepared
+        """
+        if self._data.prepared:
+            last_prepared = self._data.prepared[-1]
+            return last_prepared.view_no, last_prepared.pp_seq_no
+        return self.last_ordered_3pc
+
+    @property
+    def vc_in_progress(self):
+        return self._vc_in_progress
+
+    @property
+    def is_synced(self):
+        return self._is_synced
+
+    @property
+    def is_participating(self):
+        return self._is_participating
+
+    @vc_in_progress.setter
+    def vc_in_progress(self, vc_status: bool):
+        self._vc_in_progress = vc_status
+
+    @is_synced.setter
+    def is_synced(self, sync_status: bool):
+        self._is_synced = sync_status
+
+    @is_participating.setter
+    def is_participating(self, particip_status: bool):
+        self._is_participating = particip_status
+
+    @property
+    def h(self):
+        return self._data.h
+
+    @property
+    def H(self):
+        return self._data.H
+
+    @property
+    def last_ordered_3pc(self):
+        return self._data.view_no, self._data.pp_seq_no
+
+    def set_vc_in_progress(self, msg: ViewChangeInProgress):
+        self.vc_in_progress = msg.in_progress
+
+    def set_is_synced(self, msg: LedgerSyncStatus):
+        self.is_synced = msg.is_synced
+
+    def set_is_participating(self, msg: ParticipatingStatus):
+        self.is_participating = msg.is_participating
+
+    def has_already_ordered(self, view_no, pp_seq_no):
+        return compare_3PC_keys((view_no, pp_seq_no),
+                                self.last_ordered_3pc) >= 0
+
+    def _validate(self, msg):
```

Comment:
I think this needs to be merged with `CommonValidator`, I'll send a PR soon with initial implementation